boa/src/syntax/parser/statement/if_stm/mod.rs

```diff
@@ -68,6 +68,6 @@ impl TokenParser for IfStatement {
             _ => None,
         };
 
-        Ok(Node::if_node::<_, _, Node, _>(cond, then_stm, else_stm))
+        Ok(If::new::<_, _, Node, _>(cond, then_stm, else_stm).into())
```

Comment:
This should just return an `If`, not a `Node`.

Reply:
Done :)